src/v0/gateway.ts

```diff
@@ -68,9 +100,12 @@ export function createGateway({
         await wait(attempts ** 2 * 1000);
         return post(path, body, attempts + 1, init);
       }
-      return Promise.reject(
-        new ManifoldError({ type: ErrorType.ServerError, message: resp.statusText })
-      );
+
+      throw new ManifoldError({ type: ErrorType.ServerError, message: resp.statusText });
```

Comment:
I changed all of htese `Promise.reject` instances to `throw` which I believe is the model we're encouraged to be thinking in when using `async` functions.

Reply:
yeah, throwing an error is the same as rejecting with one. it's strange how there's multiple constructs for doing the same thing, but thats JS i guess 🤷‍♂️